crates/evm/src/evm/system_contracts/src/Bridge.sol

```diff
@@ -166,22 +166,38 @@ contract Bridge is Ownable2StepUpgradeable {
         emit OperatorUpdated(operator, _operator);
     }
     
-    /// @notice Checks if two byte sequences are equal
-    /// @dev This is not efficient, and a better approach would be doing a hash based comparison but as this is ran in a zkEVM, hashing is inefficient
+    /// @notice Checks if two byte sequences are equal in chunks of 32 bytes
+    /// @dev This approach compares chunks of 32 bytes using bytes32 equality checks for optimization
     /// @param a First byte sequence
     /// @param b Second byte sequence
     function isBytesEqual(bytes memory a, bytes memory b) internal pure returns (bool result) {
         require(a.length == b.length, "Lengths do not match");
 
-        // Cannot use keccak as its costly in ZK environment
-        uint length = a.length;
-        for (uint i = 0; i < length; i++) {
-            if (a[i] != b[i]) {
-                result = false;
-                return result;
+        uint256 fullChunks = a.length / 32;
+        for (uint256 i = 0; i < fullChunks; i++) {
+            bytes32 chunkA;
+            bytes32 chunkB;
+            assembly {
+                chunkA := mload(add(a, add(32, mul(i, 32)))) 
```

Comment:
and to even further optimize, offset can be calculated outside of for loop once, and 32 could be added on every loop instead of calculating the multiplication every time. basically instead of having i variable in the loop, you can start the offset from 32, and increment 32 in the for loop

Reply:
As this requires an additional memory load and memory store, its actually more expensive when comapred to one add and one mul. We just implemented both to test practically and it seems that this version is just a little more expensive.

Reply:
That was my guess for the reason why it was more costly. I just checked one more and even if the gas costs are super close this one (using offset) is slightly more expensive. I can share my testing code with you if you want and maybe you can try it out.